sdk/core/Azure.Core/src/Shared/AzureKeyCredentialPolicy.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core.Pipeline;
+
+namespace Azure.Core
+{
+    internal class AzureKeyCredentialPolicy : HttpPipelineSynchronousPolicy
+    {
+        private readonly string _keyHeader;
+        private readonly AzureKeyCredential _credential;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AzureKeyCredentialPolicy"/> class.
+        /// </summary>
+        /// <param name="credential">The <see cref="AzureKeyCredential"/> used to authenticate requests.</param>
+        /// <param name="keyHeader">The name of the API key header used for signing requests.</param>
+        public AzureKeyCredentialPolicy(AzureKeyCredential credential, string keyHeader)
```

Comment:
Api keys can sometimes be passed as query parameters. One such example is Azure cognitive search. (however, for azure cognitive search, you can use both a header or a query parameter interchangeable, so we can pick one). 

I don't know if we currently have services that only support query string api keys in Azure, but we had a discussion this week for a service that was considering either that or a shared access signature query parameter.

Reply:
For .NET, we are keeping this policy in a private API via shared source. People can use it or create their own logic for the policy in order to handle that scenario.
Will need to see how this will work for other languages though